### PR TITLE
Update calico to 3.18.1

### LINKED ISSUE #767: Upgrade Calico to 3.18.x
**Is your feature request related to a problem? Please describe.**
I want to use more recent Calico version as the CNI

### PR DESCRIPTION
**Pull request by soider**
- Update upstream manifests
- Update default image versions in configuration
- update get-calico.sh to work with actual yf command api
closes #767

### CODE REVIEW

File: static/manifests/calico/ConfigMap/calico-config.yaml [hunk 1]
```diff
@@ -11,11 +11,10 @@ data:
...
   # Configure the backend to use.
   calico_backend: "{{ .Mode }}"
+
   # Configure the MTU to use for workload interfaces and tunnels.
-  # - If Wireguard is enabled, set to your network MTU - 60
-  # - Otherwise, if VXLAN or BPF mode is enabled, set to your network MTU - 50
-  # - Otherwise, if IPIP is enabled, set to your network MTU - 20
-  # - Otherwise, if not using any encapsulation, set to your network MTU.
+  # By default, MTU is auto-detected, and explicitly setting this field should not be required.
+  # You can override auto-detection by providing a non-zero value.
```
Comment (jnummelin) [inline static/manifests/calico/ConfigMap/calico-config.yaml:17]: Does this mean we should now default to `0` in k0s?

Comment (soider) [reply static/manifests/calico/ConfigMap/calico-config.yaml:17]: yeah, makes sense, changed it.

Comment (jnummelin) [reply static/manifests/calico/ConfigMap/calico-config.yaml:17]: Changing this mean we need to adjust the docs too for this field.

**Review by jnummelin — CHANGES_REQUESTED** [review]
The MTU change requires small docs change too

**Review by jnummelin — APPROVED** [review]

**Review by ncopa — APPROVED** [review]

**Pull request merged**